src/utils/validateTrimmedString.ts

```diff
@@ -1,10 +1,41 @@
-// makes sure the string passed into the function isn't an empty string
-export function validateTrimmedString(input: string) {
-	const trimmedInput = input.trim();
+import { ListItem } from "../api";
 
+// Validates the item name input for a shopping list
+export function validateItemName(
+	input: string,
+	existingItems: ListItem[],
+): string | null {
+	const trimmedInput = input.trim(); //removes leading and trailing whitespaces
+
+	// Condition 1: Check if the input is empty
 	if (trimmedInput.length === 0) {
-		return null;
+		return "Item cannot be empty";
+	}
+
+	//Remove punctuation marks and normalize input
+	const punctuationRegex = /[^\p{L}]/gu;
+
+	const normalizedInputName = trimmedInput
+		.replace(punctuationRegex, "")
+		.toLowerCase();
+
+	//Create a list of  normalized existing item names
+	const normalizedExistingItemNames = existingItems.map((existingItem) => {
+		return existingItem.name.replace(punctuationRegex, "").toLowerCase();
+	});
+
+	// Condition 2: Check if the normalized input matches any existing item
+	const isDuplicateItem = (normalizedInputName: string): boolean => {
+		return normalizedExistingItemNames.some(
+			(item) => item === normalizedInputName,
+		);
+	};
+
+	//return error if the item already exists
+	if (isDuplicateItem(normalizedInputName)) {
+		return ` ${normalizedInputName} already exists in the list`;
 	}
 
-	return trimmedInput;
+	// Return null if no errors are found (input is valid)
```

Comment:
question: is there a way without breaking everything to have this return `trimmedInput` again. I added an item to a test list that was '     hi' and it did trim it and i can't add 'hi' in any different form again I did notice in the DB it is adding the item with the white space to it instead of the `trimmedInput` like before?

Reply:
https://github.com/user-attachments/assets/6c191b26-2ab5-419f-b876-4f0bf5325282

I tried to show that the toast and such does show the trimmed value but when checking the DB there is a single whitespace before the goodbye. Sorry if it is awkward to see 😅

Reply:
Thats a great point. It was something we explored. I was trying to create a piece of logic with brute force, using if - else statements to return either the error if duplicate item, or return either the normalized input name. The issue was in the addItemForm, the logic is returning the object that is being returned in the validateItemName function. So it reading the normalizedinputname as an error object. Its something I definetely want to explore more. The check for the duplicate error in the addItemForm before the item is added can be refactored..